skcosmo/utils/eig_solver.py

```diff
@@ -22,8 +23,10 @@ def eig_solver(matrix, n_components=None, tol=1e-12, add_null=False):
                      False
     :type add_null: boolean
     """
-
-    v, U = np.linalg.eig(matrix)
+    if n_components is not None and n_components < matrix.shape[0]:
+        v, U = speig(matrix, k=n_components, tol=tol)
+    else:
+        v, U = np.linalg.eig(matrix)
```

Comment:
Please rebase this file with master -- if you're not changing anything it needn't be in the PR.